docker/ray/Dockerfile

```diff
@@ -1,11 +1,23 @@
 ARG GPU
-FROM rayproject/ray-deps:nightly"$GPU"
-ARG WHEEL_PATH
-ARG FIND_LINKS_PATH=".whl"
+ARG DOCKER_PREFIX
+
+# Pull the wheel from the ray-wheel build
+FROM "$DOCKER_PREFIX"ray-wheel:nightly"$GPU" AS raywheel
+WORKDIR /ray
+
+FROM "$DOCKER_PREFIX"ray-deps:nightly"$GPU"
+WORKDIR /ray
 # For Click
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
-COPY $WHEEL_PATH .
-COPY $FIND_LINKS_PATH $FIND_LINKS_PATH
-RUN $HOME/anaconda3/bin/pip --no-cache-dir install `basename $WHEEL_PATH`[all] \
---find-links $FIND_LINKS_PATH && sudo rm `basename $WHEEL_PATH`
+# Specify bazel (see https://github.com/bazelbuild/bazel/issues/12887)
+ENV USE_BAZEL_VERSION=3.7.2
+RUN sudo apt-get update && sudo apt-get install -y curl bash
+RUN echo "export PATH=~/anaconda3/bin/:~/.cargo/bin:\$PATH" >> ~/.bashrc
+SHELL ["/bin/bash", "-c"]
+RUN export PATH=$HOME/.cargo/bin:$HOME/anaconda3/bin/:$PATH
+# Install py-spy no arm on conda yet so fall through to pypi wheel
+RUN (mamba install --yes py-spy) || (pip install --upgrade pip && pip install py-spy)
```

Comment:
Can we stick to either pulling our dependencies from pypi or just directly installing them from github (ideally pinned to a tagged commit)? It just makes it a bit easier for us to track where we're pulling dependencies in from?

Reply:
Yeah I can change this one to just always use pypi

Reply:
To be fair though, just using pypi is going to be slower because some of the packages can be installed faster from conda-forge.